cmd/dra-example-controller/driver.go

```diff
@@ -94,9 +94,17 @@ func (d driver) GetClaimParameters(ctx context.Context, claim *resourcev1.Resour
 
 func (d driver) Allocate(ctx context.Context, claim *resourcev1.ResourceClaim, claimParameters interface{}, class *resourcev1.ResourceClass, classParameters interface{}, selectedNode string) (*resourcev1.AllocationResult, error) {
 	if selectedNode == "" {
-		return nil, fmt.Errorf("TODO: immediate allocations not yet supported")
+		return d.allocateImmediateClaim(claim, claimParameters, class, classParameters)
 	}
 
+	if err := d.allocateDelayedClaim(claim, claimParameters, class, classParameters, selectedNode); err != nil {
+		return nil, err
+	}
```

Comment:
I try and avoid this style of:
```
if err := blah(); err != nil 
```
I can go into a long explanation of why, but the short answer is that it keeps it consistent with how all other calls that return an error along with other variables need to handle their errors. Since this is how it's been done in this codebase elsewhere, let's keep it that way for consistency.

Reply:
Sure, alright.